### DISCUSSION TITLE
Documentation on the query language given to `select`

### DISCUSSION THREAD

**Discussion by davetron5000** [supabase/supabase#11411] [Questions]
Hi, I'm coming to the realization that Supabase doesn't support SQL via the JS client library and it seems that the string you give to `select` has a lot of rules and is basically a language of sorts.

Is there are definitive document about how this language is defined and how it works?  Looking at [Postgrest](https://postgrest.org/en/stable/api.html#vertical-filtering-columns) documentation I can sorta see some similarities, but the HTTP stuff is much different from the string given to `select`.

I think in a perfect world there would be a way to take SQL and figure out the incantation for `select()` to produce the same results, but I realize the select mini-language may not support everything in SQL.

**Comment by GaryAustin1** [comment]
There are three places with some documentation.
The API tab on your dashboard,
The Supabase docs https://supabase.com/docs/reference/javascript/select (including filters and modifiers below that)
And then PostgREST docs like you found.  The postgres-js library converts the javascript syntax to the API REST URL format.

To use SQL you need to use rpc functions.

**Reply by davetron5000** [reply]
# API Tab

I had not seen this, and it is somewhat helpful. It's not really documentation, though, it just shows some examples.  They are helpful, though.

# Supbase docs

These are also example based and don't document the format of the string.  For example, you can alias a column by using a colon, but this is not mentioned anywhere:

```javascript
from("widgets").select("id, widgetName:widget_name")
```

# PostgREST Docs

I can sorta piece together things from this. I understand that postgres-js somehow converts the string to the REST API, but how does it do this?

What I'm looking for is more of a reference-style documentation about the string provided to `select` rather than examples. Examples are great, but when I want to dig deeper, it's not easy to put it all together.  Most of the other methods accept structured parameters and are reasonably well documented, but `select` seems to accept a somewhat magical string and just wondering how I can understand what is possible?

**Reply by GaryAustin1** [reply]
I understand.  I don't know of anything else besides looking at the source for [supabase/postgrest-js](https://github.com/supabase/postgrest-js/tree/master/src).
Maybe someone else knows of other documentation, or there is something new Supabase added last month with the new docs and I've just not run across it yet.

I normally use the PostgREST docs to get the detail for the select part of the string.  Everything after the ? in the url are filters or modifiers.

**Reply by silentworks** [reply]
The places that @GaryAustin1 has mentioned are the only places where documentation is for the `.select` method. @davetron5000 Yes this can be improved with adding more advanced features to the docs. I will bring this up with the docs team and add this to our list of tasks to get done.

**Comment by EskelCz** [comment]
Thanks a lot for this. I asked the AI in docs about the alias and it hallucinated this answer...
<img width="754" alt="alias" src="https://user-images.githubusercontent.com/1198558/234845768-72f64a0c-ffe6-4aea-a2d2-fc2c88335bb5.png">
